Dalamud/Interface/Internal/TextureManager.cs

```diff
@@ -0,0 +1,458 @@
+    /// <param name="textureManager">TextureManager instance.</param>
+    public TextureManagerPluginScoped(DataManager dataManager, TextureManager textureManager)
+    {
+        this.dataManager = dataManager;
+        this.textureManager = textureManager;
+    }
+
+    /// <inheritdoc/>
+    public IDalamudTextureWrap? GetIcon(
+        uint iconId,
+        ITextureProvider.IconFlags flags = ITextureProvider.IconFlags.ItemHighQuality,
+        ClientLanguage? language = null,
+        bool keepAlive = false)
+    {
+        var wrap = this.textureManager.GetIcon(iconId, flags, language, keepAlive);
+        if (wrap == null)
+            return null;
+        
+        this.trackedTextures.Add(wrap);
+        return wrap;
+    }
+
+    /// <inheritdoc/>
+    public IDalamudTextureWrap? GetTextureFromGamePath(string path, bool keepAlive = false)
+    {
+        var wrap = this.textureManager.GetTextureFromGamePath(path, keepAlive);
+        if (wrap == null)
+            return null;
+        
+        this.trackedTextures.Add(wrap);
+        return wrap;
+    }
+
+    /// <inheritdoc/>
+    public IDalamudTextureWrap GetTexture(TexFile file)
+    {
+        return this.dataManager.GetImGuiTexture(file) as DalamudTextureWrap ?? throw new ArgumentException("Could not load texture");
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        // Dispose all leaked textures
+        foreach (var textureWrap in this.trackedTextures.Where(x => !x.IsDisposed))
+        {
+            textureWrap.Dispose();
+        }
+    }
+}
+
+/// <summary>
+/// Wrap.
+/// </summary>
+internal class TextureManagerTextureWrap : IDalamudTextureWrap
+{
+    private readonly TextureManager manager;
+    private readonly string path;
+    private readonly bool keepAlive;
+
+    private int? width;
+    private int? height;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TextureManagerTextureWrap"/> class.
+    /// </summary>
+    /// <param name="path">The path to the texture.</param>
+    /// <param name="keepAlive">Keep alive or not.</param>
+    /// <param name="manager">Manager that we obtained this from.</param>
+    internal TextureManagerTextureWrap(string path, bool keepAlive, TextureManager manager)
+    {
+        this.path = path;
+        this.keepAlive = keepAlive;
+        this.manager = manager;
+    }
+
+    /// <inheritdoc/>
+    public IntPtr ImGuiHandle => this.manager.GetInfo(this.path).Wrap!.ImGuiHandle;
+
+    /// <inheritdoc/>
+    public int Width => this.width ??= this.manager.GetInfo(this.path).Wrap!.Width;
+
+    /// <inheritdoc/>
+    public int Height => this.height ??= this.manager.GetInfo(this.path).Wrap!.Height;
+
+    /// <summary>
+    /// Gets a value indicating whether or not this wrap has already been disposed.
+    /// If true, the handle may be invalid.
+    /// </summary>
+    internal bool IsDisposed { get; private set; }
+
+    /// <inheritdoc/>
+    public void Dispose()
```

Comment:
To make sure I understand: we should still be disposing our textures, but we have an in-place backstop on unload in case we forget?

Reply:
Yes, you should, but it will also be done for you if your plugin unloads, and the texture is unloaded if it hasn't been drawn for a while so impact is minimized if you don't.